commcare-cloud/commcare_cloud/commands/ansible/ansible_playbook.py

```diff
@@ -229,3 +235,198 @@ def run(self, args, unknown_args):
+        'pg_standby': ['postgresql']
+    }
+    ACTIONS = ['start', 'stop', 'restart', 'status']
+    DESIRED_STATE_FOR_ACTION = {
+        'start': 'started',
+        'stop': 'stopped',
+        'restart': 'restarted',
+    }
+    # add this mapping where service group is not same as the inventory group itself
+    INVENTORY_GROUP_FOR_SERVICE = {
+        'stanchion': 'stanchion',
+        'elasticsearch': 'elasticsearch',
+        'zookeeper': 'zookeeper',
+    }
+    # add this if the service package is not same as the service itself
+    SERVICE_PACKAGES_FOR_SERVICE = {
+        'rabbitmq': 'rabbitmq-server',
+        'kafka': 'kafka-server'
+    }
+    # add this if the module name is not same as the service itself
+    ANSIBLE_MODULE_FOR_SERVICE = {
+        "redis": "redis-server",
+
+    }
+
+    def make_parser(self):
+        super(Service, self).make_parser()
+        self.parser.add_argument(
+            'service_group',
+            choices=self.SERVICES.keys(),
+            help="The service group to run the command on"
+            )
+        self.parser.add_argument(
+            'action',
+            choices=self.ACTIONS,
+            help="What action to take"
+        )
+        self.parser.add_argument(
+            '--only',
+            help=(
+                "Specific comma separated services to act on for the service group. "
+                "Example Usage: --only=riak,stanchion"
+            )
+        )
+
+    def get_inventory_group_for_service(self, service, service_group):
+        return self.INVENTORY_GROUP_FOR_SERVICE.get(service, service_group)
+
+    def run_status_for_service(self, service_group, args, unknown_args):
+        for service in self.run_for_services(service_group, args):
+            if service == "redis":
+                args.shell_command = "redis-cli ping"
+            else:
+                args.shell_command = "service %s status" % self.SERVICE_PACKAGES_FOR_SERVICE.get(service, service)
+            args.inventory_group = self.get_inventory_group_for_service(service, args.service_group)
+            RunShellCommand(self.parser).run(args, unknown_args)
+
+    def run_ansible_module_for_service_group(self, service_group, args, unknown_args, inventory_group=None):
+        for service in self.SERVICES[service_group]:
+            action = args.action
+            state = self.DESIRED_STATE_FOR_ACTION[action]
+            args.inventory_group = (
+                inventory_group or
+                self.get_inventory_group_for_service(service, service_group))
+            args.module = 'service'
+            args.module_args = "name=%s state=%s" % (
+                self.ANSIBLE_MODULE_FOR_SERVICE.get(service, service),
+                state)
+            RunAnsibleModule(self.parser).run(
+                args,
+                unknown_args
+            )
+
+    def run_for_services(self, service_group, args):
+        if args.only:
+            return args.only.split(',')
```

Comment:
Would be good to have this check that it's a subset of `self.SERVICES[service_group]`. Otherwise you could pass in any random word here and it'd cause pretty unintuitive behavior or at least probably give a much worse error message.

Reply:
I see, you do that check below, in `ensure_permitted_only_options`. There might be a way to reorganize this so the control flow is clearer, but we can leave that for another day.

Reply:
@dannyroberts i could add choices when calling `add_arguments` and that would work if we do space separated arguments like `--only riak stanch` but then had to do this manual check later since we moved to comma-separated. Open to better ideas.